### PR TITLE
feat: new chart error panel

### LINKED ISSUE #1520: Downsampling is automatically applied to charts that don't support it
Newly-created plots with too many points to actually plot appear to have downsampling enabled even when it's not supported (i.e. when the X axis is not a datetime). For example, the following query:

```groovy
t = emptyTable(250_000).updateView("RowIdx = ii").update("Val = Math.random()")

plot = figure().plot("Plot", t, "RowIdx", "Val").show()
```

Creates this plot (notice the 'downsampling failed' error icon):

image

There is nothing indicating that downsampling should be used or would be required; the only thing you see is a blank chart and that downsampling failed.

When disabling downsampling, the UI isn't always responsive — the panel doesn't 'spin' or anything, so it's hard to tell that it's working. For example, when I click to disable downsampling on that plot, this is still all I see ~3 seconds later:

image

It does eventually draw (although for larger tables w/ millions of rows, it appears to hang indefinitely):

image

When downsampling or plotting fails, it would be helpful to display the number of source table rows — makes it easier as a user to understand what you're telling DH to do/what you're doing wrong.

### PR DESCRIPTION
**Pull request by ethanalvizo**
Closes #1520

### CODE REVIEW

File: packages/chart/src/Chart.tsx [hunk 1]
```diff
@@ -704,23 +708,22 @@ class Chart extends Component<ChartProps, ChartState> {
...
-          {shownError != null && (
-            <>
-              <div className="chart-error">{shownError}</div>
-              <CopyButton tooltip="Copy Error" copy={shownError}>
-                Copy Error
-              </CopyButton>
-            </>
-          )}
-        </Popper>
+        {downsamplingError != null && (
```
Comment (ethanalvizo) [inline packages/chart/src/Chart.tsx:711]: wasn't quite sure what the difference is between downsampling errors and other types of errors. Do we not want them to show in this window as well?

Currently I have them trigger their own overlay warning

**Comment by ethanalvizo** [comment]
image


New error popup^

File: packages/chart/src/ChartErrorOverlay.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+              Dismiss
+            </Button>
+          )}
+          {onConfirm && !undismissableError && (
+            <Button onClick={onConfirm} kind="primary">
+              Continue
+            </Button>
+          )}
+        </div>
+        {/* )} */}
```
Comment (mofojed) [inline packages/chart/src/ChartErrorOverlay.tsx:53]:
```suggestion
```

File: packages/chart/src/ChartErrorOverlay.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    errorMessage === 'Too many items to disable downsampling';
+
+  return (
+    <div className="chart-panel-overlay chart-error-overlay">
+      <div className="chart-panel-overlay-content chart-error-overlay-content">
+        <div className="info-message" data-testid={messageTestId}>
+          {errorMessage}
+          <CopyButton copy={errorMessage} style={{ margin: '0' }} />
+        </div>
+        {/* {!undismissableError && ( */}
```
Comment (mofojed) [inline packages/chart/src/ChartErrorOverlay.tsx:35]: Remove undismissable commented out code
```suggestion
```

File: packages/chart/src/ChartErrorOverlay.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  onCancel,
+  'data-testid': dataTestId,
+}: ChartErrorOverlayProps): ReactElement {
+  const messageTestId =
+    dataTestId != null ? `${dataTestId}-message` : undefined;
+
+  // TODO: DHC #5220 to detect slow performance and show this message
+
+  const undismissableError =
+    errorMessage === 'Too many items to disable downsampling';
```
Comment (mofojed) [inline packages/chart/src/ChartErrorOverlay.tsx:26]: Remove this `undismissableError` part.
```suggestion
```

**Review by mofojed — CHANGES_REQUESTED** [review]

**Review by mofojed — APPROVED** [review]

**Review by mofojed — APPROVED** [review]

**Pull request merged**